source/Cosmos.Core/Memory/Heap.cs

```diff
@@ -32,6 +32,40 @@ public static unsafe void Init()
             HeapLarge.Init();
         }
 
+        /// <summary>
+		/// Re-allocates or "re-sizes" data asigned to a pointer.
```

Comment:
Maybe add that this must be a pointer to the start of the allocated memory region